spec/unit/util/execution_spec.rb

```diff
@@ -219,6 +244,36 @@ def stub_process_wait(exitstatus)
           end
         end
 
+        describe "cwd option" do
+          def expect_cwd_to_be(cwd)
+            Puppet::Util::Execution.expects(executor).with(
```

Comment:
Implementation testing here. Is there a way to stub this method and instead check for: `Dir.expects(:chdir).with(cwd)` in the tests below? that way we aren't testing implementation for parameter passing.

Reply:
I don't think this is implementation testing b/c of the `anything` argument matcher.

We can't (and should not) be using `chdir` here because setting the process to run in the right `cwd` is left to the `execute_posix` and `execute_windows` (i.e. the executor) functions, respectively, as they're meant to handle any platform-specific issues with the `cwd` like different pathing styles for Windows, which will be handled in PUP-9150. Note that we use `chdir` in `Windows::Process.execute` because that works for our use-case for now, but in PUP-9150 that will likely be removed and passed as an option to `Process.create`.